Decisions.AdobeSign/Steps/DownloadDocument.cs

```diff
@@ -1,45 +1,46 @@
 ﻿using Decisions.AdobeSign.Utility;
+using Decisions.OAuth;
 using DecisionsFramework.Design.ConfigurationStorage.Attributes;
 using DecisionsFramework.Design.Flow;
 using DecisionsFramework.Design.Flow.Mapping;
 using DecisionsFramework.Design.Properties;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using DecisionsFramework.ServiceLayer.Services.ContextData;
 
 namespace Decisions.AdobeSign.Steps
 {
-    [AutoRegisterStep("Download Document", adobeSignCategory)]
+    [AutoRegisterStep("Download Document", STEP_PARAMS_CATEGORY)]
     [Writable]
     public class DownloadDocument : AbstractStep
     {
+        private const string RESULT_PATH = "Done";
+        private const string INPUT_NAME_FILE_PATH = "File Path";
+        private const string INPUT_NAME_AGREEMENT_ID = "Agreement Id";
+
         [PropertyHidden]
-        public override DataDescription[] InputData
-        {
-            get
-            {
-                var data = new DataDescription[] { new DataDescription(typeof(string), AbstractStep.AgreementIdLabel), new DataDescription(typeof(string), AbstractStep.FilePathLabel), };
-                return base.InputData.Concat(data).ToArray();
-            }
-        }
+        public override DataDescription[] InputData => new DataDescription[] 
+        { 
+            new (typeof(string), INPUT_NAME_AGREEMENT_ID), 
+            new (typeof(string), INPUT_NAME_FILE_PATH)
+        };
 
-        public override OutcomeScenarioData[] OutcomeScenarios
+        protected override OutcomeScenarioData[] GetOutcomeScenarios()
         {
-            get
+            return new OutcomeScenarioData[]
             {
-                var data = new OutcomeScenarioData[] { new OutcomeScenarioData(resultOutcomeLabel) };
-                return base.OutcomeScenarios.Concat(data).ToArray();
-            }
+                new (RESULT_PATH)
+            };
         }
-
-        protected override Object ExecuteStep(AdobeSignConnection conn, StepStartData data)
+        
+        protected override ResultData ExecuteStep(StepStartData data, OAuthToken token)
         {
-            string agreementId = (string)data.Data[AbstractStep.AgreementIdLabel];
-            string filePath = (string)data.Data[AbstractStep.FilePathLabel];
-            AdobeSignApi.GetTransientDocument(conn, agreementId, filePath);
-            return null;
+            string agreementId = (string)data.Data[INPUT_NAME_AGREEMENT_ID];
+            AdobeSignApi.GetTransientDocument(
+                token, 
+                agreementId, 
+                filePath: (string)data.Data[INPUT_NAME_FILE_PATH]);
+            return new ResultData(
+                resultPath: RESULT_PATH, 
+                values: new [] { new DataPair(RESULT_PATH, agreementId) });
```

Comment:
I think this is trying to return some data even though our outcome path has no output data. I don't think this will actually change the behavior of the step, so I'm going to approve.